ideascube/mediacenter/management/commands/import_medias.py

```diff
@@ -93,21 +93,38 @@ def add(self, metadata):
                                 title)
             return
 
-        path = os.path.join(self.ROOT, original)
-        if not os.path.isfile(path):
-            self.report.error(u'Path not found', path)
-            return
-        with open(path, 'rb') as f:
-            original = File(f, name=original)
+        original_path = os.path.join(self.ROOT, original)
+        preview = metadata.get('preview')
+        preview_path = os.path.join(self.ROOT, preview) if preview else None
+        of = pf = None
+        try:
+            try:
+                of = open(original_path, 'rb')
+            except OSError as e:
+                self.report.error("Cannot open path", e)
 
-            preview = metadata.get('preview')
             if preview:
-                path = os.path.join(self.ROOT, preview)
-                with open(path, 'rb') as f:
-                    preview = File(f, name=preview)
-                    self.save(metadata, original, instance, preview)
+                try:
+                    pf = open(preview_path, 'rb')
+                except OSError as e:
+                    self.report.error("Cannot open path", e)
+
+            if not of:
+                return
+            if preview and not pf:
+                return
+
+            original = File(of, name=original)
+            if pf:
+                preview = File(pf, name=preview)
+                self.save(metadata, original, instance, preview)
             else:
                 self.save(metadata, original, instance)
```

Comment:
Put `preview = None` in the `else` block, and then you can call `self.save(metadata, original, instance, preview=preview)` outside of the `if`/`else`?